common/djangoapps/third_party_auth/tests/specs/test_testshib.py

```diff
@@ -155,41 +203,154 @@ def test_configure_testshib_provider_with_cache_duration(self):
-                self.assertEqual(num_total, 1)
+    # Note that these details are different than those that will be provided by the SAML
+    # assertion metadata. Rather, they will be fetched from the mocked SAPSuccessFactors API.
+    USER_EMAIL = "john@smith.com"
+    USER_NAME = "John Smith"
+    USER_USERNAME = "jsmith"
 
-    def do_provider_login(self, provider_redirect_url):
-        """ Mocked: the user logs in to TestShib and then gets redirected back """
-        # The SAML provider (TestShib) will authenticate the user, then get the browser to POST a response:
-        self.assertTrue(provider_redirect_url.startswith(TESTSHIB_SSO_URL))
-        return self.client.post(
-            self.complete_url,
-            content_type='application/x-www-form-urlencoded',
-            data=self.read_data_file('testshib_response.txt'),
+    def setUp(self):
+        super(SuccessFactorsIntegrationTest, self).setUp()
+        # Mock out httpretty calls for the SuccessFactors API
+
+        # Mock the call to the SAP SuccessFactors assertion endpoint
+        SAPSF_ASSERTION_URL = 'http://successfactors.com/oauth/idp'
+
+        def assertion_callback(_request, _uri, headers):
+            """
+            Return a fake assertion after checking that the input is what we expect.
+            """
+            self.assertIn('private_key=fake_private_key_here', _request.body)
+            self.assertIn('user_id=myself', _request.body)
+            self.assertIn('token_url=http%3A%2F%2Fsuccessfactors.com%2Foauth%2Ftoken', _request.body)
+            self.assertIn('client_id=TatVotSEiCMteSNWtSOnLanCtBGwNhGB', _request.body)
+            return (200, headers, 'fake_saml_assertion')
+
+        httpretty.register_uri(httpretty.POST, SAPSF_ASSERTION_URL, content_type='text/plain', body=assertion_callback)
+
+        SAPSF_BAD_ASSERTION_URL = 'http://successfactors.com/oauth-fake/idp'
+
+        def bad_callback(_request, _uri, headers):
+            """
+            Return a 404 error when someone tries to call the URL.
+            """
+            return (404, headers, 'NOT AN ASSERTION')
+
+        httpretty.register_uri(httpretty.POST, SAPSF_BAD_ASSERTION_URL, content_type='text/plain', body=bad_callback)
+
+        # Mock the call to the SAP SuccessFactors token endpoint
+        SAPSF_TOKEN_URL = 'http://successfactors.com/oauth/token'
+
+        def token_callback(_request, _uri, headers):
+            """
+            Return a fake assertion after checking that the input is what we expect.
+            """
+            self.assertIn('assertion=fake_saml_assertion', _request.body)
+            self.assertIn('company_id=NCC1701D', _request.body)
+            self.assertIn('grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Asaml2-bearer', _request.body)
+            self.assertIn('client_id=TatVotSEiCMteSNWtSOnLanCtBGwNhGB', _request.body)
+            return (200, headers, '{"access_token": "faketoken"}')
+
+        httpretty.register_uri(
```

Comment:
@haikuginger Nit: Change formatting to match other calls to `register_uri` in this method.